daemon/src/model/cfd.rs

```diff
@@ -284,17 +287,50 @@ pub enum CfdState {
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 pub struct Attestation {
     pub id: OracleEventId,
-    pub price: u64,
     pub scalars: Vec<SecretKey>,
+    #[serde(with = "::bdk::bitcoin::util::amount::serde::as_sat")]
+    payout: Amount,
+    price: u64,
 }
 
-impl From<oracle::Attestation> for Attestation {
-    fn from(attestation: oracle::Attestation) -> Self {
-        Attestation {
-            id: attestation.id,
-            price: attestation.price,
-            scalars: attestation.scalars,
-        }
+impl Attestation {
+    pub fn new(
+        id: OracleEventId,
+        price: u64,
+        scalars: Vec<SecretKey>,
+        dlc: Dlc,
+        role: Role,
+    ) -> Result<Self> {
+        let cet = dlc
+            .cets
+            .iter()
+            .find_map(|(_, cet)| cet.iter().find(|cet| cet.range.contains(&price)))
+            .context("Unable to find attested price in any range")?;
+
+        let payout = cet
+            .tx
+            .output
+            .iter()
+            .find_map(|output| {
+                (output.script_pubkey
+                    == match role {
+                        Role::Maker => dlc.maker_address.script_pubkey(),
+                        Role::Taker => dlc.taker_address.script_pubkey(),
+                    })
+                .then(|| Amount::from_sat(output.value))
```

Comment:
Can we break this up a bit so it is easier to read? I have to turn my head side ways for this :joy:

Reply:
:D